misk/src/main/kotlin/misk/client/OkHttpClientCommonConfigurator.kt

```diff
@@ -0,0 +1,66 @@
+package misk.client
+
+import com.google.inject.Inject
+import com.google.inject.Singleton
+import okhttp3.ConnectionPool
+import okhttp3.Dispatcher
+import okhttp3.OkHttpClient.Builder
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+@Singleton
+class OkHttpClientCommonConfigurator @Inject internal constructor() {
+  fun configure(builder: Builder, config: HttpClientEndpointConfig): Builder {
```

Comment:
Should we inject `Provider<HttpClientsConfig>` so we don't need to pass `HttpClientEndpointConfig` as an argument?

Reply:
Unfortunately [HttpClientFactory.get](https://github.com/podlech/misk/blob/ce105346cf587e473ef5e46c3335ad3235e2cab5/misk/src/main/kotlin/misk/client/HttpClientFactory.kt#L28) receives the config as a parameter, so I'll keep it as is.